### PR TITLE
[RFC] vim-patch:7.4.1799, 7.4.1806, 7.4.1808, 7.4.1809

### PR DESCRIPTION
**Pull request by Shougo**
patch 7.4.1799

Problem:    'guicolors' is a confusing option name.
Solution:   Use 'termguicolors' instead. (Hirohito Higashi)

https://github.com/vim/vim/commit/61be73bb0f965a895bfb064ea3e55476ac175162

patch 7.4.1806

Problem:    'termguicolors' option missing from the options window.
Solution:   Add the entry.

https://github.com/vim/vim/commit/8e3d1b6326c103cc92f8d07b1161ee5172acf201

patch 7.4.1808

Problem:    Using wrong feature name to check for 'termguicolors'.
Solution:   Use the right feature name. (Ken Takata)

https://github.com/vim/vim/commit/8a24b794b89916c8074892e7b25121a21f1fa9c9

patch 7.4.1809

Problem:    Using wrong short option name for 'termguicolors'.
Solution:   Use the option name.

https://github.com/vim/vim/commit/868cfc19bb079a16ca58884b551486566f35419b

### CODE REVIEW

**Comment by Shougo** [comment]
Nothing comments?

File: src/nvim/option_defs.h [hunk 1]
```diff
@@ -619,6 +619,7 @@ EXTERN long p_titlelen;         ///< 'titlelen'
 EXTERN char_u *p_titleold;      ///< 'titleold'
 EXTERN char_u *p_titlestring;   ///< 'titlestring'
 EXTERN char_u *p_tsr;           ///< 'thesaurus'
+EXTERN bool p_tgc;               ///< 'termguicolors'
```
Comment (justinmk) [inline src/nvim/option_defs.h]: alignment?

**Comment by justinmk** [comment]
Merged after a small formatting change, and a doc addendum.

It only works if the setting is enabled at startup (e.g. `nvim --cmd 'set termguicolors'` or in init.vim). But not with `nvim -c`, nor if `:set termguicolors` is invoked after startup. I am guessing this doesn't match the Vim behavior. So I mentioned this in the doc.

**Comment by Shougo** [comment]
Thank you!

**Pull request closed without merging**